cluster-autoscaler/simulator/nodes_test.go

```diff
@@ -17,175 +17,175 @@ limitations under the License.
 			},
 		},
-		Spec: apiv1.PodSpec{
-			NodeName: nodeName,
-		},
-	}
-
-	podDaemonsetAnnotation := &apiv1.Pod{
-		ObjectMeta: metav1.ObjectMeta{
-			Namespace:       "default",
-			Name:            "podDaemonset2",
-			SelfLink:        "podDaemonset2",
-			OwnerReferences: GenerateOwnerReferences("ds2", "CustomDaemonset", "crd/v1", ""),
-			Annotations: map[string]string{
-				pod_util.DaemonSetPodAnnotationKey: "true",
+		{
+			name: "node with a deleted mirror pod",
+			node: test.BuildTestNode("n", 1000, 10),
+			pods: []*apiv1.Pod{
+				test.SetMirrorPodSpec(test.BuildScheduledTestPod("p1", 100, 1, "n")),
+				func() *apiv1.Pod {
+					p := test.SetMirrorPodSpec(test.BuildScheduledTestPod("p2", 100, 1, "n"))
+					p.DeletionTimestamp = &now
+					return p
+				}(),
+			},
+			wantPods: []*apiv1.Pod{
+				test.SetMirrorPodSpec(test.BuildScheduledTestPod("p1", 100, 1, "n")),
 			},
 		},
-		Spec: apiv1.PodSpec{
-			NodeName: nodeName,
-		},
-	}
-
-	tests := []struct {
-		name      string
-		inputPods []*apiv1.Pod
-		want      []*apiv1.Pod
-	}{
 		{
-			name:      "nil input pod list",
-			inputPods: nil,
-			want:      []*apiv1.Pod{},
+			name: "node with a DS pod [forceDS=false]",
+			node: test.BuildTestNode("n", 1000, 10),
+			pods: []*apiv1.Pod{
+				test.SetDSPodSpec(test.BuildScheduledTestPod("p1", 100, 1, "n")),
+			},
+			wantPods: []*apiv1.Pod{
+				test.SetDSPodSpec(test.BuildScheduledTestPod("p1", 100, 1, "n")),
+			},
 		},
 		{
-			name:      "should return only mirrorPod",
-			inputPods: []*apiv1.Pod{pod1, mirrorPod},
-			want:      []*apiv1.Pod{mirrorPod},
+			name: "node with a DS pod [forceDS=true, no daemon sets]",
+			node: test.BuildTestNode("n", 1000, 10),
+			pods: []*apiv1.Pod{
+				test.SetDSPodSpec(test.BuildScheduledTestPod("p1", 100, 1, "n")),
+			},
+			forceDS: true,
 		},
 		{
-			name:      "should ignore podDeleted",
-			inputPods: []*apiv1.Pod{pod1, mirrorPod, podDeleted},
-			want:      []*apiv1.Pod{mirrorPod},
+			name: "node with a DS pod [forceDS=true, some daemon sets]",
+			node: test.BuildTestNode("n", 1000, 10),
+			pods: []*apiv1.Pod{
+				test.SetDSPodSpec(test.BuildScheduledTestPod("p1", 100, 1, "n")),
+			},
+			daemonSets: []*appsv1.DaemonSet{ds1, ds2},
+			forceDS:    true,
+			wantPods: []*apiv1.Pod{
+				func() *apiv1.Pod { p := daemon.NewPod(ds1, "n"); p.Name = "ds1"; return p }(),
+				func() *apiv1.Pod { p := daemon.NewPod(ds2, "n"); p.Name = "ds2"; return p }(),
+			},
 		},
 		{
-			name:      "should return daemonset pod",
-			inputPods: []*apiv1.Pod{pod1, podDaemonset},
-			want:      []*apiv1.Pod{podDaemonset},
+			name: "everything together [forceDS=false]",
+			node: test.BuildTestNode("n", 1000, 10),
+			pods: []*apiv1.Pod{
+				test.BuildScheduledTestPod("p1", 100, 1, "n"),
+				test.BuildScheduledTestPod("p2", 100, 1, "n"),
+				test.SetMirrorPodSpec(test.BuildScheduledTestPod("p3", 100, 1, "n")),
+				func() *apiv1.Pod {
```

Comment:
nit: I'd consider introducing a named helper function instead of in-lining functions in test cases, especially when multiple cases do the same thing (e.g. add deletion timestamp). Ideally, I'd love to have some builders for test pods and nodes, but that's clearly out of scope here.

Reply:
Done